CIP-0048/CIP-0048.md

```diff
@@ -0,0 +1,141 @@
+
+
+Specifically, this proposal aims to solve for the following:
+
+* Provide a mechanism to reduce duplicated metadata into one or multiple transactions that can be referenced
+* Provide a mechanism that allows on chain data to be referenced
+* Provide a mechanism to reference on chain data in a set order
+* Provide a mechanism to reference on chain data stored in a different policy
+
+# Specification
+Please note I offer optional single char names to allow futher reduction in metadata size if desired
+
+*Inside the 721 metadata tag*
+|reserved 721 metadata tag |description | example
+|---|---| --- |
+|references, refs, r | contains data tag and src tag | ```'r':{ "data:"text/plain","src":[0,1] }```|
+|payload, p | contains raw data | ```'p':{ "0":["h","e","l","l","o"], "1":"world" }``` |
+|version| this is described in 721 | not sure if it should be version 1.1? instead of 2|
+
+*inside the refences tag*
+|tags that are reseved inside references |description | example
+|---|---| --- |
+|data| contains the data type MIME | for example if base 64, data:"text/plain;base64"|
+|src|contains a list of poitners to payloads | ```"src":[0,1]``` or ``` "src":["payload0","payload1"]```|
+|policy, p | references another policy. If none, assume current policy | ```'r':{"data:"text/plain","src":{[0,1]},"p":"<POLICY_ID>}"```|
+|txhash, tx | references a specific tx hash (optional) | ```'r':{"data:"text/plain","src":{[0,1]},"tx":"<TX_HASH>}"```|
+|type | the format/type of the data inside the payload |```'r':{"data:"image/jpeg","src":{[0,1]},"type":"ipfs"```|
+|b, back, backup | fallback, if a link to the data is used in the 'src' tag and is broken revert to a fallback link. | ```'r':{"src":{[0,1]},"b":{[2,3]}} ``` or for multiple fallbacks use ```'r':{"src":{[0,1]},"b":"{'0':{[2,3]}, '1'{[4,5]} }"``` |
+
+*defintion of the different reference types*
+| type | desciption |
+|---| ---|
+|ipfs|data is stored at ipfs|
+|url|data is found a url|
+|api|data is found at api get request|
+|utf-8| data is directly in payload as utf-8 |
+|b64 | data is in base 64 encoded in payload |
+|reference? | advance feature, points to a reference |
+|none| assume data is raw and using the defined media type |
+
+*?references in payloads could cause issues with circular dependencies, maybe this should be more like inode in ext2 or the depth defined*
+
+
+```
+{
+   "721":{
+      "<POLICY_ID>":{
+         "<NFT_NAME_B16>":{
+            "project":"<PROJECT>",
+            "name":"<NFT_NAME>",
+            "references":{
+               "type":"raw"
+               "data":"text/plain"
+               "src":[
+                  0,
+                  1
+               ]
+            }
+         },
+         "payload":{
+            "0":["Hello"],"1":["World!"]
+         },
+         "version":"2"
+      }
+   }
+}
+```
+# Retrieve payload information using references
+1. Find all transactions with the 721 label
```

Comment:
What is the scope of research here? A block? The whole chain? Should only payload under the same policy be considered?